modules/apps/depot/depot-web/src/main/resources/META-INF/resources/screen/navigation/entries/sites.jsp

```diff
@@ -33,12 +33,21 @@ List<DepotEntryGroupRel> depotEntryGroupRels = depotAdminSitesDisplayContext.get
 			<span class="heading-text"><liferay-ui:message key="connected-sites" /></span>
 		</clay:content-col>
 
+		<%
+		Map<String, Object> additionalProps = new HashMap<>();
+
+		additionalProps.put("currentURL", currentURL);
+		additionalProps.put("eventName", liferayPortletResponse.getNamespace() + "selectSite");
+		additionalProps.put("itemSelectorURL", String.valueOf(depotAdminSitesDisplayContext.getItemSelectorURL()));
+		%>
+
 		<clay:content-col>
 			<span class="heading-end">
 				<clay:button
+					additionalProps="<%= additionalProps %>"
 					displayType="secondary"
-					id='<%= renderResponse.getNamespace() + "addConnectedSiteButton" %>'
 					label="add"
+					propsTransformer="js/addConnectedSiteButtonPropsTransformer"
```

Comment:
Last concern here would be how this affects Time to Interactive. 

In this case, the click listener was already registered lazily, so there's no change in behaviour. Button will be unresponsive (do nothing) until JS runs.

Not sure what would be the best UX for this... chances are the user won't notice on a fast enough connection or with a good enough preloading heuristic, so optimizing here might be an overkill. One option would be to pass `disabled=true` here and then `disabled=false` in the `propsTransformer`, but not sure if that's better.

In other cases where we have plain `<button onclick="window.foo()">` and we define `foo` in a no-dependencies `script`, we would transforming it to the previous case. How comfortable are we with this?

Reply:
In the past I worked with a system that had a delegated event listener on the document (a tiny snippet of JS in the `<head>`), and there would be `data-something` attributes on server-rendered buttons, links and other elements. If a button/link etc got clicked on before the JS had loaded, it would "do the right thing" (if possible! Back then, things were simpler... it often amounted to just an AJAX POST or GET and inserting the response into the DOM). Another approach I seem to recall trying is having the listener effectively capture the event and replay it once the JS was there. And yet another is just to have a kind of progressive enhancement where clicking does some "natural" action (like submitting a form or triggering a navigation); works for simple cases, but not for complex UIs (eg. buttons that open dropdowns or modals that have to fetch their contents etc).

From a UX perspective, I think people are fine with seeing disabled buttons that only become active when a page is fully loaded. It beats having a button that lets you click it but does nothing.

> One option would be to pass `disabled=true` here and then `disabled=false` in the `propsTransformer`, but not sure if that's better.

I was going to say the gotcha with this approach is that if the user has JS disabled then the page won't work at all... but it seems that ship has pretty much sailed for DXP, right?

Reply:
> In the past I worked with a system that had a delegated event listener on the document (a tiny snippet of JS in the <head>), and there would be data-something attributes on server-rendered buttons, links and other elements. If a button/link etc got clicked on before the JS had loaded, it would "do the right thing" (if possible! Back then, things were simpler... it often amounted to just an AJAX POST or GET and inserting the response into the DOM). Another approach I seem to recall trying is having the listener effectively capture the event and replay it once the JS was there.

I'm pretty sure `Liferay.provide` was introduced to handle these kinds of things.

> but it seems that ship has pretty much sailed for DXP, right?

![I hope so](https://user-images.githubusercontent.com/156388/85026836-90e9de00-b14f-11ea-87d9-21bc3ff35774.gif)

Reply:
Yep, just ask @julien. He's been removing many `Liferay.provide` calls because we want to be able to turn off AUI by default, and it turns out that with a little bit of refactoring, in many, many cases, we didn't actually need the proxying that `Liferay.provide`, er, provides. But it's not surprising that there may be some cases where you still want something along those lines. We just figured we wouldn't try to make a new (non-AUI) equivalent of `Liferay.provide` yet until we found a case where we couldn't do without it (haven't found one yet... 🤞  — sure we will eventaully though).

Reply:
BTW I'm fine with disabling it until JS is loaded. But we might want hear what a UX designer has to say about it

Reply:
> [...] We just figured we wouldn't try to make a new (non-AUI) equivalent  [...]

Sorry to disappoint you, @wincent, but we actually did a while ago... [`Liferay-lazyLoad`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/lazy_load.js#L16) 😂

Reply:
Deprecate `Liferay.lazyLoad`!

Long live `Liferay.lazyLoad2`!

Reply:
😂 

So... for now buttons (dropdown menus) will just do nothing until hydrated like it is right now... we can easily improve the UX later by defaulting to `disabled` when in need of hydration later if we find this troubling.